graphql_compiler/compiler/emit_cypher.py

```diff
@@ -108,11 +124,16 @@ def _emit_with_clause_components(cypher_steps):
     if not cypher_steps:
         return []
 
-    result = [u'WITH']
-    location_names = {
-        cypher_step.as_block.location.get_location_name()[0]
-        for cypher_step in cypher_steps
-    }
+    result = []
+    location_names = set()
+    for cypher_step in cypher_steps:
+        location = cypher_step.as_block.location
+        if isinstance(location, Location):
+            location_name = location.get_location_name()[0]
+        else:
+            # must be a FoldScopeLocation because as_block is a MarkLocation block.
```

Comment:
How sure are you about this? If you are very sure, then assert it: `elif isinstance(location, FoldScopeLocation):` + `else: raise AssertionError(<message about unreachable code here>)`. If you aren't very sure, then don't claim it in a comment :)